### DISCUSSION TITLE
Any possible support for kotlin DSL query and mutation?

### DISCUSSION THREAD

**Discussion by ronjunevaldoz** [ExpediaGroup/graphql-kotlin#1405] [Ideas]
Example DSL for query and mutation. 

```kotlin
 // reset password
query("resetPassword") {
    description = "Generates a new password and send via email."
    resolver { code: String, loginLink: String ->
        userUseCase.resetPassword(code, loginLink)
    }
}
// forgot password
mutation("forgotPassword") {
    description = "Generates a reset password link with code and send via email."
    resolver { emailOrUsername: String, resetPasswordLink: String ->
        userUseCase.forgotPassword(emailOrUsername, resetPasswordLink)
    }
}
``` 

Source: https://github.com/aPureBase/KGraphQL

**Comment by dariuszkuc** [comment]
Hello 👋 
We believe that using DSL to define is only useful in some very simple use cases (basically RPC calls that you would traditionally expose as part of controller) and quickly shows its limitations. With our code first approach you would simply wrote those as functions accepting some parameters.

GraphQL already provides a schema definition language (SDL) that is used to define the schema and is much more flexible than the DSL approach. SDL-first is a pretty common approach for designing your GraphQL schemas, e.g. it is the default approach supported by the `graphql-java` (which we rely on for query execution).
